File: web/src/types/greenhouse-harvest-seed.ts
/** How much greenhouse seed was sown for a greenhouse harvest — the input side, recorded while
 * planning. */
export type GreenhouseHarvestSeed = {
  id: number;
  greenhouseHarvestId: number;
  greenhouseSeedId: number;
  amount: number;
};

export type GreenhouseHarvestSeedInput = Omit<GreenhouseHarvestSeed, 'id'>;

File: server/Models/GreenhouseHarvestSeed.cs
namespace Server.Models;

/// <summary>
/// How much of a <see cref="GreenhouseSeed"/> was sown for a <see cref="GreenhouseHarvest"/> — the
/// input side, recorded while planning. The greenhouse counterpart to <see cref="HarvestSeed"/>.
/// Adding one deducts the amount from the seed; there is no movement ledger for greenhouse seed,
/// so unlike the field version this doesn't also log a history row.
/// </summary>
public class GreenhouseHarvestSeed
{
    public int Id { get; set; }
    public int GreenhouseHarvestId { get; set; }
    public int GreenhouseSeedId { get; set; }
    public decimal Amount { get; set; }
}
